pkg/marathon/client_test.go

```diff
@@ -16,3 +23,73 @@ func TestResponseToError(t *testing.T) {
 	err = httpResponseToError(&resp)
 	assert.EqualError(t, err, `HTTP 404 error`)
 }
+
+func TestAddAppWithEmptyInput(t *testing.T) {
+	client := &Client{API: &marathonmocks.MarathonMock{}}
+	ctx := mock.NewContext(&cli.Environment{Input: strings.NewReader("")})
+
+	newApp, err := client.AddApp(ctx, "")
+	assert.Error(t, err, nil)
+	assert.Nil(t, newApp)
+}
+
+func TestAddAppWithNonExistingFile(t *testing.T) {
+	client := &Client{API: &marathonmocks.MarathonMock{}}
+	ctx := mock.NewContext(nil)
+
+	newApp, err := client.AddApp(ctx, "/this/does/not/exist")
+	assert.Error(t, err, nil)
+	assert.Nil(t, newApp)
+}
+
+func TestAddApp(t *testing.T) {
+	tests := []struct {
+		name             string
+		applicationByErr error
+		applicationByApp *goMarathon.Application
+		expectedApp      *goMarathon.Application
+		expectedErr      error
+	}{
+		{
+			name:             "application does not exists, yet",
+			applicationByErr: &goMarathon.APIError{ErrCode: goMarathon.ErrCodeNotFound},
+			applicationByApp: nil,
+			expectedApp:      &goMarathon.Application{ID: "some id"},
+			expectedErr:      nil,
+		},
+		{
+			name:             "arbitrary APIError",
+			applicationByErr: &goMarathon.APIError{ErrCode: 999},
```

Comment:
Please add one more test wiht error that is not `goMarathon.APIError` to check all cases